enumerables.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  # rubocop:disable Style/For
+  def my_each
+    return to_enum unless block_given?
```

Comment:
let us improve this method by passing the name of the method as an argument to ```to_enum(:my_each)```
 thus 
``` return to_enum(method_name) unless block_given? ```  instead of  ``` return to_enum unless block_given? ```. Apply this in other instances where an enumerator ought to be returned when no block is given, thus ```my_each```, ```my_each_with_index```, ```my_select``` and ```my_map```